itegration: husqvarna_automower_ble: Initial commit

alistair23 opened the pull request:
## Proposed change

This PR adds documentation for the Husqvarna Automower BLE integration.

The Husqvarna Automower BLE integration provides connectivity with Husqvarna Automowers lawn mowers via a local Bluetooth connection. This allows connecting and controlling an Automower without any accounts, cloud or network connection.

The integration is based on [AutoMower-BLE](https://github.com/alistair23/AutoMower-BLE), a unofficial reverse engineered Husqvarna Automower Connect BLE library.

## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [X] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [X] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/108326
- Link to parent pull request in the Brands repository: https://github.com/home-assistant/brands/pull/6032
- This PR fixes or closes issue: fixes #

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [X] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [X] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

- **New Features**
	- Introduced documentation for integrating Husqvarna Automower BLE lawn mowers with Home Assistant.
	- Added step-by-step setup instructions for connecting and controlling Automowers via Bluetooth.
	- Supported device types include Lawn Mower and Battery Level.
	- Integration categorized under Lawn Mower in Home Assistant, with local polling and configuration flow support.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

alistair23 committed:
itegration: husqvarna_automower_ble: Initial commit

Signed-off-by: Alistair Francis <alistair@alistair23.me>

alistair23 commented:
As people have been asking for this feature and this PR seems to have stalled. If you would like to manually install this component you can follow these steps

### Add version to manifest

You need to add a version to the manifest, you can apply this diff

```diff
diff --git a/homeassistant/components/husqvarna_automower_ble/manifest.json b/homeassistant/components/husqvarna_automower_ble/manifest.json
index cb3ee2a502..bd7db5f1f1 100644
--- a/homeassistant/components/husqvarna_automower_ble/manifest.json
+++ b/homeassistant/components/husqvarna_automower_ble/manifest.json
@@ -12,5 +12,6 @@
   "dependencies": ["bluetooth_adapters"],
   "documentation": "https://www.home-assistant.io/integrations/???",
   "iot_class": "local_polling",
-  "requirements": ["automower-ble==0.1.32"]
+  "requirements": ["automower-ble==0.1.32"],
+  "version": "0.1.0"
 }
```

Then just rsync the entire `homeassistant/components/husqvarna_automower_ble` directory from this PR to `/root/homeassistant/custom_components/` on your HA instance

You will also want to enable debug logging by adding this to your `homeassistant/configuration.yaml` file. That helps identify any issues.

```
logger:
  # default: debug
  logs:
    custom_components.husqvarna_automower_ble: debug
    custom_components.husqvarna_automower_ble.config_flow: debug
    automower_ble.mower: debug
```

Then reboot.

After that it should work. If you have any issues please file an issue on either the HA GitHub instance or at https://github.com/alistair23/AutoMower-BLE. Please include the logs of any issues `cat ./config/home-assistant.log | grep mow` is generally a good place to start.

If this works for you maybe comment here, that might help convince someone to review this

wlan4you commented:
Hello,
we wrote an email together a few days ago.
I can hardly wait until it is hopefully firmly integrated into HA.
Unfortunately I have no skills to integrate it.
My new RASPI 5 8GB runs with Zigbee2MQTT, Masiquito, Home Asstistant Tasmota and so on in a docker system.
11 Zigbee components running in Node Red and HA.
Is it worth the wait because it is then totally easy to integrate?
Or is there a guide for beginners?

Does a device have to run with ESP Home?
The RASPI 5 can do that too, can't it?

Fayrewood commented:
Hi Alistair,
I have replaced by ESP32 with a new one with an external antenna and have copied over my YAML, which includes the BLE MAC address of the mower. 

Do I need to clear anything in HA, or just go through the pairing process again and re-add the integration?

al31c0 commented:
> Hi Alistair, I have replaced by ESP32 with a new one with an external antenna and have copied over my YAML, which includes the BLE MAC address of the mower.
> 
> Do I need to clear anything in HA, or just go through the pairing process again and re-add the integration?
> 
> My ESP is reporting as follows, but I cannot see the custom_component when I try and add the integration and I can't see anything in the logs for 'mow' or 'ble'. I have cleared the previous custom_components folder and copied the latest version over, but cannot see the component as an integration.
> 
> ```
> [17:39:01][I][esp32_ble_client:067]: [3] [0C:EC:80:E0:64:EE] 0x00 Attempting BLE connection
> [17:39:01][D][esp32_ble_client:110]: [3] [0C:EC:80:E0:64:EE] ESP_GATTC_CONNECT_EVT
> [17:39:01][D][esp32_ble_client:110]: [3] [0C:EC:80:E0:64:EE] ESP_GATTC_OPEN_EVT
> [17:39:01][D][esp32_ble_client:110]: [3] [0C:EC:80:E0:64:EE] ESP_GATTC_SEARCH_CMPL_EVT
> [17:39:01][I][esp32_ble_client:227]: [3] [0C:EC:80:E0:64:EE] Connected
> [17:39:01][D][ble_client:058]: All clients established, services released
> [17:39:01][D][esp32_ble_client:188]: [3] [0C:EC:80:E0:64:EE] cfg_mtu status 0, mtu 65
> ```

Can you help me installing the integration? Thank you so much https://github.com/alistair23/AutoMower-BLE/issues/43

Fayrewood commented:
HI @alistair23 
I have replaced my ESP32 with a new model and external antenna, but I have a couple of issues.  Could you guide me?

I cannot get the Integration working again in HA.  I have removed it from custom_components and re-added the latest version.  I didn't have a manifest.json file, so have created one, but when I try to add the integration it says it needs manual configuration.  Have I missed something from my manifest file?

```
{   
    "domain": "husqvarna_automower_ble",
    "name": "Husqvarna Automower BLE",
    "dependencies": ["bluetooth_adapters"],
    "documentation": "https://www.home-assistant.io/integrations/???",
    "iot_class": "local_polling",
    "requirements": ["automower-ble==0.1.32"],
    "version": "0.1.0"
}
```

Also, do I need to re-do anything in my ESP YAML?  I note that the instructions say to not have a ble_client entry, but it isn't clear whether this gets automatically populated when pairing or not.

```
esp32_ble_tracker:
  scan_parameters:
    active: false
    continuous: false
    duration: 50sec

bluetooth_proxy:
  active: true
  cache_services: true

ble_client:
  - mac_address: 0C:EC:80:E0:64:EE
    id:  automower_305
    on_disconnect:
      then:
        - esp32_ble_tracker.start_scan:
```

Many thanks!

jkaberg commented:
> As people have been asking for this feature and this PR seems to have stalled. If you would like to manually install this
> ....
> If this works for you maybe comment here, that might help convince someone to review this

And for the lazy people (like myself 😆)

Oneliner, be inside the `custom_components` folder when executing this

```curl -LJO https://github.com/alistair23/homeassistant-core/archive/refs/heads/alistair/husqvarna_automower_ble.zip && unzip -q homeassistant-core-alistair-husqvarna_automower_ble.zip && mv homeassistant-core-alistair-husqvarna_automower_ble/homeassistant/components/husqvarna_automower_ble ./husqvarna_automower_ble && rm homeassistant-core-alistair-husqvarna_automower_ble.zip && rm -rf homeassistant-core-alistair-husqvarna_automower_ble```

Fayrewood commented:
FABULOUS!!  Thank You!

I've got a manifest file which I was missing now!

> 
> And for the lazy people (like myself 😆)
> 
> Oneliner, be inside the `custom_components` folder when executing this
> 
> `curl -LJO https://github.com/alistair23/homeassistant-core/archive/refs/heads/alistair/husqvarna_automower_ble.zip && unzip -q homeassistant-core-alistair-husqvarna_automower_ble.zip && mv homeassistant-core-alistair-husqvarna_automower_ble/homeassistant/components/husqvarna_automower_ble ./husqvarna_automower_ble && rm homeassistant-core-alistair-husqvarna_automower_ble.zip && rm -rf homeassistant-core-alistair-husqvarna_automower_ble`

Fayrewood commented:
I am really struggling to get my mower paired after replacing my ESP32.  What I do is:

1. Power on the mower (A305) and put it into pairing mode.
2. Power on the ESP32 BT Proxy
3. Add the MAC address into the Integration config flow windows

From my early experiences, PIN security is turned off and I have tried pairing several times.

I get the following errors - any ideas?

```
[15:15:02][I][esp32_ble_client:067]: [3] [0C:EC:80:E0:64:EE] 0x00 Attempting BLE connection
[15:15:02][D][esp32_ble_client:110]: [3] [0C:EC:80:E0:64:EE] ESP_GATTC_CONNECT_EVT
[15:15:02][D][esp32_ble_client:110]: [3] [0C:EC:80:E0:64:EE] ESP_GATTC_OPEN_EVT
[15:15:02][D][esp32_ble_client:110]: [3] [0C:EC:80:E0:64:EE] ESP_GATTC_SEARCH_CMPL_EVT
[15:15:02][I][esp32_ble_client:227]: [3] [0C:EC:80:E0:64:EE] Connected
[15:15:02][D][ble_client:058]: All clients established, services released
[15:15:02][D][esp32_ble_client:188]: [3] [0C:EC:80:E0:64:EE] cfg_mtu status 0, mtu 65
[15:15:20][D][sensor:094]: 'BT Irrigation WiFi dB': Sending state -46.00000 dBm with 0 decimals of accuracy
[15:15:20][D][sensor:094]: 'BT Irrigation WiFi Percent': Sending state 100.00000 Signal % with 0 decimals of accuracy
[15:15:23][I][bluetooth_proxy:282]: [0] [0C:EC:80:E0:64:EE] Connecting v3 without cache
[15:15:23][I][esp32_ble_client:067]: [0] [0C:EC:80:E0:64:EE] 0x00 Attempting BLE connection
[15:15:23][D][esp-idf:000]: W (58533) BT_GATT: gatt_connect wrong state 4

[15:15:23][D][esp32_ble_client:110]: [0] [0C:EC:80:E0:64:EE] ESP_GATTC_OPEN_EVT
[15:15:23][D][esp32_ble_client:110]: [0] [0C:EC:80:E0:64:EE] ESP_GATTC_SEARCH_CMPL_EVT
[15:15:23][I][esp32_ble_client:227]: [0] [0C:EC:80:E0:64:EE] Connected
[15:15:23][D][esp32_ble_client:188]: [0] [0C:EC:80:E0:64:EE] cfg_mtu status 0, mtu 65
[15:15:23][D][esp32_ble_client:110]: [0] [0C:EC:80:E0:64:EE] ESP_GATTC_READ_CHAR_EVT
[15:15:24][D][esp32_ble_client:110]: [0] [0C:EC:80:E0:64:EE] ESP_GATTC_READ_CHAR_EVT
[15:15:24][D][esp32_ble_client:110]: [0] [0C:EC:80:E0:64:EE] ESP_GATTC_READ_CHAR_EVT
[15:15:24][D][esp32_ble_client:110]: [0] [0C:EC:80:E0:64:EE] ESP_GATTC_READ_CHAR_EVT
[15:15:24][D][esp32_ble_client:110]: [0] [0C:EC:80:E0:64:EE] ESP_GATTC_READ_CHAR_EVT
[15:15:24][W][bluetooth_proxy.connection:076]: [0] [0C:EC:80:E0:64:EE] Error reading char/descriptor at handle 0x E, status=5
[15:15:24][D][esp32_ble_client:110]: [0] [0C:EC:80:E0:64:EE] ESP_GATTC_READ_CHAR_EVT
[15:15:24][D][esp32_ble_client:110]: [0] [0C:EC:80:E0:64:EE] ESP_GATTC_READ_CHAR_EVT
[15:15:24][I][esp32_ble_client:084]: [0] [0C:EC:80:E0:64:EE] Disconnecting.
[15:15:24][D][esp32_ble_client:110]: [0] [0C:EC:80:E0:64:EE] ESP_GATTC_CLOSE_EVT
```

Fayrewood reviewed (approved):
Works as expected, without issue.

Fayrewood commented:
I commented out the ble_client in the ESP32 yaml and paired successfully.  Now removed comments, so that it will force a scan on disconnect.

al31c0 commented:
> > As people have been asking for this feature and this PR seems to have stalled. If you would like to manually install this
> > ....
> > If this works for you maybe comment here, that might help convince someone to review this
> 
> And for the lazy people (like myself 😆)
> 
> Oneliner, be inside the `custom_components` folder when executing this
> 
> `curl -LJO https://github.com/alistair23/homeassistant-core/archive/refs/heads/alistair/husqvarna_automower_ble.zip && unzip -q homeassistant-core-alistair-husqvarna_automower_ble.zip && mv homeassistant-core-alistair-husqvarna_automower_ble/homeassistant/components/husqvarna_automower_ble ./husqvarna_automower_ble && rm homeassistant-core-alistair-husqvarna_automower_ble.zip && rm -rf homeassistant-core-alistair-husqvarna_automower_ble`

Can you help me installing the integration? Thank you so much https://github.com/alistair23/AutoMower-BLE/issues/43

Fayrewood commented:
> Can you help me installing the integration? Thank you so much 


I can try, but I've had issues and happy to try and help.

Firstly, have you got your Bluetooth proxy ESP32 set up? If not, I can share my ESP32 code. Comment out anything you have under ble_client for now (we'll need it later).

Secondly, go to a terminal session in HA via the SSH / Terminal add on and change directory to the ./config/custom_components folder.  Delete anything you may currently have for husqvarna-Automower and rub the curl command as above to copy the correct files.

Find the MAC address of the mower which is easiest done with an android device, or an app on IOS device to scan for BT devices.

Then, edit the manifest file and add a version entry at the end of the file:
"version": "0.1.0" and save the file.

Check that the mower security level is set to low (no pin required other than power on) and put it into New Pairing mode from the accessories menu.

Restart HA and go to Integrations and ADD.  You should see husqvarna BLE as a new integration.  With the mower still in pairing mode, add an entry and enter the Mac address with the colons between each address pair.

It's worth having a second window open to view the ESP logs as it's connecting.  You should see green entries for success messages and yellow for any issues.

Back in the integration, your mower should be connected and you'll see two entries for the device name and the battery level.

If that hasn't worked, try the pairing again, paying attention to the logs in the ESP window.

If it has worked, remove the comments from the ESP code for the ble_client so that in the event of a disconnect, it will force a scan and reconnect.

Hope this help you get further.

joostlek commented:
Brands PR is incorrect. Please open a new one where you move the current logos to the husqvarna brand and then symlink them to the 2 core integrations

c0ffeeca7 commented on source/_integrations/husqvarna_automower_ble.markdown:
```diff
@@ -0,0 +1,46 @@
+This has been tested against the following mowers, although it is expected that all will work
+ - Automower 305
+
+### Home Assistant
```
```suggestion
### Prerequisites
```

c0ffeeca7 commented on source/_integrations/husqvarna_automower_ble.markdown:
```diff
@@ -0,0 +1,46 @@
+
+### Home Assistant
+
+1. Setup a [Bluetooth controller](https://www.home-assistant.io/integrations/bluetooth/), an ESPHome Bluetooth proxy works well and allows locating a device close to the mower.
```
```suggestion
1. Setup a [Bluetooth controller](https://www.home-assistant.io/integrations/bluetooth/). An ESPHome Bluetooth proxy works well and allows locating a device close to the mower.
```

c0ffeeca7 commented on source/_integrations/husqvarna_automower_ble.markdown:
```diff
@@ -0,0 +1,46 @@
+
+1. Setup a [Bluetooth controller](https://www.home-assistant.io/integrations/bluetooth/), an ESPHome Bluetooth proxy works well and allows locating a device close to the mower.
+
+2. Enter pairing mode on the mower. Different models will do this in different ways. For the 305 for example the mower will enter pairing mode for the first 3 minutes after power on. Ensure the mower is in pairing mode when adding the integration. This only needs to be done once per BLE controller (so changing the ESPHome device will require a repair).
```
```suggestion
2. Enter pairing mode on the mower. Different models will do this in different ways. For the 305, for example, the mower will enter pairing mode for the first 3 minutes after powering on. Ensure the mower is in pairing mode when adding the integration. This only needs to be done once per BLE controller (so changing the ESPHome device will require a repair).
```

c0ffeeca7 commented on source/_integrations/husqvarna_automower_ble.markdown:
```diff
@@ -0,0 +1,46 @@
+
+2. Enter pairing mode on the mower. Different models will do this in different ways. For the 305 for example the mower will enter pairing mode for the first 3 minutes after power on. Ensure the mower is in pairing mode when adding the integration. This only needs to be done once per BLE controller (so changing the ESPHome device will require a repair).
+
+3. Add the integration to your Home Assistant installation:
+   [![my_button](https://my.home-assistant.io/badges/config_flow_start.svg)](https://my.home-assistant.io/redirect/config_flow_start/?domain=husqvarna_automower_ble)
+
```
```suggestion
```
Removing this, as it is part of the config_flow

c0ffeeca7 commented on source/_integrations/husqvarna_automower_ble.markdown:
```diff
@@ -0,0 +1,46 @@
+3. Add the integration to your Home Assistant installation:
+   [![my_button](https://my.home-assistant.io/badges/config_flow_start.svg)](https://my.home-assistant.io/redirect/config_flow_start/?domain=husqvarna_automower_ble)
+
+4. When prompted you will need to enter the mower BLE Mac address. You can find this in the ESPHome logs, an Android phone or some other means
```
```suggestion
4. When adding the integration to Home Assistant, you will need to enter the mower BLE Mac address. You can find this in the ESPHome logs, on an Android phone, or by some other means.
```

c0ffeeca7 commented on source/_integrations/husqvarna_automower_ble.markdown:
```diff
@@ -0,0 +1,46 @@
+   [![my_button](https://my.home-assistant.io/badges/config_flow_start.svg)](https://my.home-assistant.io/redirect/config_flow_start/?domain=husqvarna_automower_ble)
+
+4. When prompted you will need to enter the mower BLE Mac address. You can find this in the ESPHome logs, an Android phone or some other means
+
+5. Confirm successful connection of mower and assign to an area.
```
```suggestion
```

c0ffeeca7 commented on source/_integrations/husqvarna_automower_ble.markdown:
```diff
@@ -0,0 +1,46 @@
+
+5. Confirm successful connection of mower and assign to an area.
+
+Pairing can take a few goes. Even when using the official Android application it can be tricky to get the first pair to succeed. If you are having issues reboot the mower and try again.
```
```suggestion
Pairing can take a few goes. Even when using the official Android application, it can be tricky to get the first pair to succeed. If you are having issues, reboot the mower and try again.
```

c0ffeeca7 commented on source/_integrations/husqvarna_automower_ble.markdown:
```diff
@@ -0,0 +1,41 @@
+
+2. Enter pairing mode on the mower. Different models will do this in different ways. For the 305, for example, the mower will enter pairing mode for the first 3 minutes after powering on. Ensure the mower is in pairing mode when adding the integration. This only needs to be done once per BLE controller (so changing the ESPHome device will require a repair).
+
+4. When adding the integration to Home Assistant, you will need to enter the mower BLE Mac address. You can find this in the ESPHome logs, on an Android phone, or by some other means.
```
```suggestion
3. When adding the integration to Home Assistant, you will need to enter the mower BLE Mac address. You can find this in the ESPHome logs, on an Android phone, or by some other means.
```

c0ffeeca7 commented on source/_integrations/husqvarna_automower_ble.markdown:
```diff
@@ -0,0 +1,41 @@
+
+### Prerequisites
+
+1. Setup a [Bluetooth controller](https://www.home-assistant.io/integrations/bluetooth/). An ESPHome Bluetooth proxy works well and allows locating a device close to the mower.
+
+2. Enter pairing mode on the mower. Different models will do this in different ways. For the 305, for example, the mower will enter pairing mode for the first 3 minutes after powering on. Ensure the mower is in pairing mode when adding the integration. This only needs to be done once per BLE controller (so changing the ESPHome device will require a repair).
+
+3. When adding the integration to Home Assistant, you will need to enter the mower BLE Mac address. You can find this in the ESPHome logs, on an Android phone, or by some other means.
```
```suggestion
1. Setup a [Bluetooth controller](https://www.home-assistant.io/integrations/bluetooth/). An ESPHome Bluetooth proxy works well and allows locating a device close to the mower.
2. Enter pairing mode on the mower. Different models will do this in different ways. For the 305, for example, the mower will enter pairing mode for the first 3 minutes after powering on. Ensure the mower is in pairing mode when adding the integration. This only needs to be done once per BLE controller (so changing the ESPHome device will require a repair).
3. When adding the integration to Home Assistant, you will need to enter the mower BLE Mac address. You can find this in the ESPHome logs, on an Android phone, or by some other means.
```

c0ffeeca7 commented on source/_integrations/husqvarna_automower_ble.markdown:
```diff
@@ -0,0 +1,39 @@
+
+The integration is based on [AutoMower-BLE](https://github.com/alistair23/AutoMower-BLE), a unofficial reverse engineered Husqvarna Automower Connect BLE library.
+
+### Tested an verified mowers
```
```suggestion
### Tested and verified mowers
```

c0ffeeca7 commented on source/_integrations/husqvarna_automower_ble.markdown:
```diff
@@ -0,0 +1,39 @@
+ha_domain: husqvarna_automower_ble
+---
+
+The Husqvarna Automower BLE integration provides connectivity with Husqvarna Automowers lawn mowers via a local Bluetooth connection. This allows connecting and controlling an Automower without any accounts, cloud or network connection.
```
```suggestion
The Husqvarna Automower BLE integration provides connectivity with Husqvarna Automowers lawn mowers via a local Bluetooth connection. This allows connecting and controlling an Automower without any accounts, cloud, or network connection.
```

c0ffeeca7 commented on source/_integrations/husqvarna_automower_ble.markdown:
```diff
@@ -0,0 +1,39 @@
+
+### Tested and verified mowers
+
+This has been tested against the following mowers, although it is expected that all will work
```
```suggestion
It is expected that all Husqvarna mowers will work. However, the integration has been tested against the following mowers:
```

c0ffeeca7 commented on source/_integrations/husqvarna_automower_ble.markdown:
```diff
@@ -0,0 +1,39 @@
+### Prerequisites
+
+1. Setup a [Bluetooth controller](https://www.home-assistant.io/integrations/bluetooth/). An ESPHome Bluetooth proxy works well and allows locating a device close to the mower.
+2. Enter pairing mode on the mower. Different models will do this in different ways. For the 305, for example, the mower will enter pairing mode for the first 3 minutes after powering on. Ensure the mower is in pairing mode when adding the integration. This only needs to be done once per BLE controller (so changing the ESPHome device will require a repair).
```
```suggestion
2. Enter the pairing mode on the mower. Different models will do this in different ways. For the 305, for example, the mower will enter pairing mode for the first 3 minutes after powering on. Ensure the mower is in pairing mode when adding the integration. This only needs to be done once per BLE controller (so changing the ESPHome device will require a repair).
```

c0ffeeca7 commented on source/_integrations/husqvarna_automower_ble.markdown:
```diff
@@ -0,0 +1,39 @@
+### Tested and verified mowers
+
+It is expected that all Husqvarna mowers will work. However, the integration has been tested against the following mowers:
+ - Automower 305
```
```suggestion

  - Automower 305
```

c0ffeeca7 commented on source/_integrations/husqvarna_automower_ble.markdown:
```diff
@@ -0,0 +1,40 @@
+
+{% include integrations/config_flow.md %}
+
+There is currently support for the following device types within Home Assistant:
+
+- Lawn Mower
+- Battery Level
```
```suggestion
```

c0ffeeca7 commented on source/_integrations/husqvarna_automower_ble.markdown:
```diff
@@ -0,0 +1,40 @@
+
+The Husqvarna Automower BLE integration provides connectivity with Husqvarna Automowers lawn mowers via a local Bluetooth connection. This allows connecting and controlling an Automower without any accounts, cloud, or network connection.
+
+The integration is based on [AutoMower-BLE](https://github.com/alistair23/AutoMower-BLE), an unofficial reverse engineered Husqvarna Automower Connect BLE library.
```
```suggestion
The integration is based on [AutoMower-BLE](https://github.com/alistair23/AutoMower-BLE), an unofficial reverse engineered Husqvarna Automower Connect BLE library.

There is currently support for the following device types within Home Assistant:

- Lawn Mower
- Battery Level
```

c0ffeeca7 commented:
Thank you for contributing, @alistair23 :tada: 

I took some liberties with the docs and restructured a bit. Could you take a look and check, if it is still correct?

alistair23 commented:
> Brands PR is incorrect. Please open a new one where you move the current logos to the husqvarna brand and then symlink them to the 2 core integrations

Updated! New PR is here: https://github.com/home-assistant/brands/pull/6032

alistair23 commented:
> Thank you for contributing, @alistair23 🎉
> 
> I took some liberties with the docs and restructured a bit. Could you take a look and check, if it is still correct?

Thanks!

My GitHub page wasn't updating, so I missed this and addressed your comments manually. This should still include the improvements and restructures you were looking for.

c0ffeeca7 reviewed (approved):
Thank you, @alistair23 :+1:
